core/src/main/scala/org/apache/spark/scheduler/AsyncEventQueue.scala

```diff
@@ -136,7 +139,9 @@ private class AsyncEventQueue(
    */
   private[scheduler] def stop(): Unit = {
     if (!started.get()) {
-      throw new IllegalStateException(s"Attempted to stop $name that has not yet started!")
+      throw new SparkIllegalStateException(
+        errorClass = "SCHEDULER_ASYNC_STOP",
+        messageParameters = Map("name" -> name.toString))
     }
```

Comment:
We can avoid these sort of off error classes - particularly for cases which are not due to user input, but due to internal state of spark.

Reply:
Do you suggest using an INTERNAL_ERROR_SCHEDULER class for this instead? There are similar INTERNAL_ERROR classes used for core

Reply:
Hi @mridulm should I do this?